autosklearn/ensemble_builder.py

```diff
@@ -475,7 +478,7 @@ def __init__(
         self.precision = precision
         self.memory_limit = memory_limit
         self.read_at_most = read_at_most
-        self.random_state = check_random_state(random_state)
+        self.random_state = random_state
```

Comment:
Don't you have to update the docstring and type annotation if you drop the `check_random_state`?

Reply:
It is updated. All objects accepting random_state arguments are of the type `random_state: Optional[Union[int, np.random.RandomState]] = None`.

[`check_random_state`](https://github.com/scikit-learn/scikit-learn/blob/2beed5584/sklearn/utils/validation.py#L926) will always result in a `RandomState` object which we do not want. We want to keep it as the type passed in and simply forward that to the underlying estimator which will then convert it if required. If we convert it ourselves, we end up the component not being idempotent.

`check_random_state`
* Converts `None` to default `RandomState` object
* Converts `int` to a seeded `RandomState` object
* Does nothing to a `RandomState` object, leaving it as the same `RandomState` object